### DISCUSSION TITLE
Created a "Coming Soon" section for Plex users using Trailarr

### DISCUSSION THREAD

**Discussion by alexkahler** [nandyalu/trailarr#47] [Show and tell]
Hey,

Just wanted to show a cool way to use Trailarr in an unconventional way. Plex users of my server now have a small "Coming Soon" section where they can see trailers for movies that will soon be available on the server.
![image](https://github.com/user-attachments/assets/a6f371dd-24c1-40ea-a639-e265fd44e7b3)

Here's how to set it up:

- Point Trailarr's mount path from the Movies folder (on the Radarr side) to a Trailers folder (on the Trailarr side).
![image](https://github.com/user-attachments/assets/1c54e08a-70ad-4296-b4d8-f60405b2a722)
- Unmonitor all movies, except unreleased/upcoming ones. Since I always unmonitor movies once they've been downloaded in Radarr, I chose to "Sync" the Trailarr connection, ensuring that movies on Trailarr have the same status.
![image](https://github.com/user-attachments/assets/50346778-76a5-4b9c-ac85-aee5c3c49f45)
- Create a library on Plex using the "Other Videos" library type, and point it to the Trailers folder.

A good tip: Enter the release date for each movie in the trailer's "Originally Available" metadata. This way, users can sort trailers chronologically by release date. This requires a bit of manual work but is well worth it.

![image](https://github.com/user-attachments/assets/e2a873eb-98c1-40e1-be22-2bfcfcfb99ed)

Finally, Trailarr is a bit "download happy" and indiscriminate when it picks the trailer on YouTube, which sometimes leads to fan-made or unofficial cough garbage trailers being downloaded—especially for movies without a set release date (e.g., Avengers Movie No. 6357 or generic Marvel Movie No. 7432). The only way to avoid this is to turn off trailer monitoring completely or unmonitor movies without official trailers yet.

**Comment by almulder** [comment]
Why not add a plex Library as Movies and lable it commign soon

**Reply by alexkahler** [reply]
Because then Plex may mark the actual movie as "Watched"

**Comment by mrainone7p** [comment]
Isn't Trailarr expecting nested folders with the same name of each movie? When I point it to a trailers folder, it gives me an error lig saying I don't have a folder with the corresponding movie name.

**Reply by nandyalu** [reply]
Trailarr gets its metadata from Radarr/Sonarr which does require a separate folder for each media item. So Trailarr will look for media folder and if it's not found, it will fail.

**Reply by mrainone7p** [reply]
> Trailarr gets its metadata from Radarr/Sonarr which does require a separate folder for each media item. So Trailarr will look for media folder and if it's not found, it will fail.

Thanks for the follow up. So guess my question is how do you set it up so that trailarr saves the the trailers to a separate library as in the case of the coming soon library. Right now I have it so that radarr sonarr download to my movies and shows library folders, but with a coming soon section assume that trailarr would need to see those library folders but save the trailer in a separate trailers folder which can be set up as a coming soon library.

**Reply by nandyalu** [reply]
You could setup a completely different instance of Radarr which only contains coming soon movies, and enable the option in Radarr to create empty movie folder.

Or you could add a new root folder in Radarr and map that to a new library in Plex for coming soon. Also enable the empty folder creation.

Make sure you disable `wait for media` option in Trailarr Settings

**Reply by mrainone7p** [reply]
Thanks for the reply! and maybe I don't understand how this setup works but from what i understand Radarr needs to create a folder for the coming soon movie so that Trailarr can download a trailer for it in that folder. If I point a radarr instance to any folder (Library), it will be used to download all my movies and map all movies folder to it, so how would this work as a coming soon library if its downloading all trailers to that library. Unless the idea is to manually add each coming soon movie to a new radarr instance, or manually add the folders or trailers there. 

Thanks for the clarification not sure i this would be a unique use case but If I could propose a feature request that Trailarr can add monitored but not downloaded movies to a new library/folder than the default library and deletes the trailer once the availability threshold is met.

**Reply by nandyalu** [reply]
> If I point a radarr instance to any folder (Library), it will be used to download all my movies and map all movies folder to it

Not really! You can add multiple root folders in Radarr and it will ask you which root folder to use when adding a movie.

> i this would be a unique use case but If I could propose a feature request that Trailarr can add monitored but not downloaded movies to a new library/folder than the default library and deletes the trailer once the availability threshold is met.

I like the idea, but it would make things super complicated and confusing, as not all users might need/want this option and have the setup to make it work. So, it's better to use the workarounds for the few people that want the option.

**Comment by quadcom** [comment]
This is exactly what I've been trying to get set up when I fumbled on this project.

The problem I find with Plex is that it will not show the movie in a library unless the movie is downloaded, even though the movie folder is created and the trailer and cover art are present. If it can't see the movie file itself, it won't list anything.

That being said, I would like to start a vote for adding this as a built-in function/feature. I came across a project that lets you create a custom library in Plex for Tube Archivist, so any downloads you add to that app automatically pop up in Plex with all the metadata that TA has. 

https://github.com/tubearchivist/tubearchivist-plex

With that, Trailarr could create an agent for Plex that can pull the data from. Trailarr could populate the metadata with the various release dates of the Film/TV Show, etc.
